l10n_br_stock_account/wizard/stock_invoice_onshipping.py

```diff
@@ -16,28 +16,49 @@ class StockInvoiceOnShipping(models.TransientModel):
     _inherit = 'stock.invoice.onshipping'
 
     @api.multi
+    def _fiscal_doc_ref_selection(self):
+        return FISCAL_DOC_REF
+
+    @api.multi
     def _compute_fiscal_doc_ref(self):
-        ref_id = False
-        picking_obj = self.env['stock.picking']
-        for record in picking_obj.browse(
-                self._context.get('active_ids', False)):
-            move = record.move_lines[0]
-            if move.origin_returned_move_id:
-                ref_id = self.env['account.invoice'].search([
-                    ('nfe_access_key', '=',
-                     move.origin_returned_move_id.picking_id.
-                     fiscal_document_access_key)
-                ], limit=1).id
-            res = 'account.invoice,%d' % ref_id
-            return res
+        """
+        Delegate calculation of fiscal_doc_ref default.
+
+        A string argument to fields.Reference(default=...) is interpreted as
+        the actual default value instead of method to look up the default.
+
+        Here we delegate to self._fiscal_doc_ref_default() so inheriting models
+        can override it.
+        """
+        return self._fiscal_doc_ref_default()
+
+    def active_picking_ids(self):
+        context = self.env.context
+        return self.env['stock.picking'].browse(
+            context.get('active_ids') or context.get('active_id', False)
+        )
+
+    @api.model
+    def get_returned_picking_ids(self):
+        return self.active_picking_ids().mapped(
+            'move_lines.origin_returned_move_id.picking_id'
+        )
+
+    @api.multi
+    def _fiscal_doc_ref_default(self):
+        return_picking_ids = self.get_returned_picking_ids()
+        ref_id = return_picking_ids.mapped('invoice_ids')[:1].id
```

Comment:
> Não seria melhor:
>     `ref_id = return_picking_ids.mapped('invoice_ids').ids[:1]`

Não, pois `.ids` é uma lista que, nesse caso, poderia ser vazia, ter um elemento, ou ter vários elementos, ao passo que `recordset[:1]` é sempre um recordset, que pode ser vazio, ou ter um elemento.

Se `recordset[:1]` for vazio, `ref_id = recordset[:1].id` vai ser `False` o que significa que a interpolação:

    res = 'account.invoice,%d' % ref_id

Terá como resultado:

    account.invoice,0

Que é exatamente o que queremos nesse caso. Por outro lado, se pegássemos `.ids[:1]` teríamos uma lista que pode estar vazia, ou conter um elemento, e seria necessário um `if` (ou no mínimo um `or`) a mais para distinguir os dois casos e pegar o id correto.

> ou apenas `.id`?

Se o recordset resultante de `return_picking_ids.mapped('invoice_ids')` tiver múltiplos elementos, então `return_picking_ids.mapped('invoice_ids').id` vai levantar uma exceção:

    except_orm: ('ValueError', 'Expected singleton: account.invoice(x, y, z,...)')

Ao passo que `return_picking_ids.mapped('invoice_ids')[0].id` vai levantar outra exceção se `return_picking_ids.mapped('invoice_ids')` estiver vazio:

    IndexError: tuple index out of range

A solução atual é a única que não envolve nenhuma condicional.